### PR TITLE
Revamp Update package's unit tests (1/2)

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

- [x] move both stores (and their mocks) into own files
- [x] replace proto helpers with shared ones
- [x] rework Registry's Find() tests
- [x] rework Registry's HasOutgoingMessages() tests
- [x] rework Registry's FindOrCreate() tests
- [x] remove duplicate `TestStorageErrorWhenLookingUpCompletedOutcome` (already part of `Find` tests)
- [x] remove duplicate `TestUpdateRemovalFromRegistry` (already part of `FindOrCreate` tests) 
- [x] rework Registry's Send() tests
- [x] rework Registry's RejectUnprocessed() tests
- [x] add test for Registry's FailoverVersion()
- [x] add test for Registry's NewRegistry() 
- [x] remove `TestUpdateAccepted_WorkflowCompleted` (already covered) 
- [x] add test for Registry's TryResurrect()

## Why?

To make sure the update package tests are in tip top shape.

## How did you test it?

All tests (and production code changes are purely cosmetic).

### CODE REVIEW

File: service/history/workflow/update/registry.go [hunk 1]
```diff
@@ -378,7 +375,7 @@ func (r *registry) remover(id string) updateOpt {
 	)
 }
 
-func (r *registry) checkLimits(ctx context.Context) error {
```
Comment (stephanos) [inline service/history/workflow/update/registry.go:381]: Removed since Context isn't used.

File: service/history/workflow/update/registry.go [hunk 2]
```diff
@@ -234,32 +226,35 @@ func (r *registry) TryResurrect(ctx context.Context, acptOrRejMsg *protocolpb.Me
...
 		reqMsg = updMsg.GetRejectedRequest()
+	default:
 		// Ignore all other message types.
 	}
 	if reqMsg == nil {
 		return nil, nil
 	}
 	reqAny, err := anypb.New(reqMsg)
 	if err != nil {
-		return nil, invalidArgf("unable to unmarshal request body: %v", err)
```
Comment (stephanos) [inline service/history/workflow/update/registry.go:262]: This is marshalling, not unmarhalling.

File: service/history/workflow/update/registry.go [hunk 3]
```diff
@@ -161,8 +155,6 @@ func WithTracerProvider(t trace.TracerProvider) Option {
 	}
 }
 
-var _ Registry = (*registry)(nil)
```
Comment (stephanos) [inline service/history/workflow/update/registry.go:164]: `var`s are expected at the top of the file, I believe.

File: service/history/workflow/update/registry.go [hunk 4]
```diff
@@ -101,14 +103,6 @@ type (
 		FailoverVersion() int64
 	}
 
-	// Store represents the update package's requirements for reading updates from the store.
-	Store interface {
-		VisitUpdates(visitor func(updID string, updInfo *persistencespb.UpdateInfo))
-		GetUpdateOutcome(ctx context.Context, updateID string) (*updatepb.Outcome, error)
-		GetCurrentVersion() int64
-		IsWorkflowExecutionRunning() bool
-	}
```
Comment (stephanos) [inline service/history/workflow/update/registry.go:110]: This is a separate file now.

File: service/history/workflow/update/registry_test.go [hunk 5]
```diff
@@ -36,685 +37,806 @@ import (
...
+	"go.temporal.io/server/common/testing/testvars"
 	"go.temporal.io/server/service/history/consts"
 
 	persistencespb "go.temporal.io/server/api/persistence/v1"
+	. "go.temporal.io/server/common/testing/protoutils"
 	"go.temporal.io/server/internal/effect"
 	"go.temporal.io/server/service/history/workflow/update"
 )
 
-type mockUpdateStore struct {
```
Comment (stephanos) [inline service/history/workflow/update/registry_test.go:46]: This is in a separate file now.

File: service/history/workflow/update/update_test.go [hunk 6]
```diff
@@ -1158,20 +1156,5 @@ func TestCompletedWorkflow(t *testing.T) {
...
-func mustUnmarshalBody[T proto.Message](t *testing.T, protocolMsg *protocolpb.Message) T {
-	t.Helper()
-	pb := new(T)
-	ppb := reflect.ValueOf(pb).Elem()
-	pbNew := reflect.New(reflect.TypeOf(pb).Elem().Elem())
-	ppb.Set(pbNew)
-
-	require.NoError(t, protocolMsg.Body.UnmarshalTo(*pb))
-	return *pb
-}
```
Comment (stephanos) [inline service/history/workflow/update/update_test.go:1177]: Removed as we already have these helpers in a shared package.

File: service/history/workflow/update/update_test.go [hunk 7]
```diff
@@ -1158,20 +1108,5 @@ func TestCompletedWorkflow(t *testing.T) {
...
+// TODO: test aborting an Update in various states
+// TODO: test "should not be able to completed update for completed workflow"
```
Comment (stephanos) [inline service/history/workflow/update/update_test.go:1112]: These will be addressed in the next PR.

File: service/history/workflow/update/update.go [hunk 8]
```diff
@@ -45,36 +44,11 @@ import (
...
 )
 
-type (
-	// EventStore is the interface that an Update needs to read and write events
-	// and to be notified when buffered writes have been flushed. It is the
-	// expectation of this code that writes to EventStore will return before the
-	// data has been made durable. Callbacks attached to the EventStore via
-	// OnAfterCommit and OnAfterRollback *must* be called after the EventStore
-	// state is successfully written or is discarded.
-	EventStore interface {
```
Comment (stephanos) [inline service/history/workflow/update/update.go:55]: Moved to the `store.go` file now.

File: service/history/workflow/update/update_test.go [hunk 9]
```diff
@@ -54,62 +52,6 @@ const (
 	testSequencingEventID int64 = 2203
 )
 
-func successOutcome(t *testing.T, s string) *updatepb.Outcome {
-	return &updatepb.Outcome{
-		Value: &updatepb.Outcome_Success{
-			Success: payloads.EncodeString(t.Name() + s),
-		},
-	}
-}
```
Comment (stephanos) [inline service/history/workflow/update/update_test.go:63]: This was moved to the bottom.

File: service/history/workflow/update/registry.go (hunk 3 above)
Comment (alexshtin) [reply service/history/workflow/update/registry.go:164]: Generally yes, but not for this type of check. I prefer it to be after type declaration.

File: service/history/workflow/update/registry.go (hunk 2 above)
Comment (alexshtin) [reply service/history/workflow/update/registry.go:262]: I think I renamed it in few places already.

**Review by alexshtin — APPROVED** [review]
image
it's just I am not sure about the colour.

**Pull request merged**